### ISSUE TITLE
Please help test 1.2.10-SNAPSHOT

### ISSUE THREAD

**Issue by pjdarton** [jenkinsci/docker-plugin#904]
**TL;DR:** If this issue is still open when you read this, please try out the [latest](https://ci.jenkins.io/job/Plugins/job/docker-plugin/job/master/) (unreleased) build and comment here to tell me if it works for you.

### Full explanation:
Folks,
We have a bunch of changes that have been [merged but not yet released](https://github.com/jenkinsci/docker-plugin/blob/master/CHANGELOG.md#unreleased).
In the past, I'd put the not-yet-released build on my own (production) Jenkins and use it (for a week or two) to make sure that nothing catches fire ... but I can't do that anymore. I no longer have access to a non-trivial Jenkins that uses the docker-plugin (my own usage of this plugin is now "local only" - I'm not using it for non-trivial stuff anymore - and that's not really a "real world" test).

...but I'd still like the plugin to get some "real world" testing before I push out a new release, hence this call for help.

The following folks have all been involved in conversations here on github in all the issues that have been addressed since the last release (1.2.9), so hopefully you'll all be sufficiently motivated to help on this: @Bitals , @CJCombrink , @Crosse , @bguerin , @dchsueh , @stephan48 , plus I'll mention @dnwe and @fraz3alpha who I know are likely well placed for this kind of thing 😉 
...but all assistance is welcome.

So, can (some of) you download a pre-release (as described in the changelog) and upload it into a "real world" Jenkins and check that nothing explodes as a result. If there are issues, mention them here first (we can create new issues & PRs if necessary).
**Note:** You _will_ need to have docker-java-api-plugin 3.2.x from now on, so make sure you aren't still on 3.1.x before making the jump.

Once I've got adequate confirmation that everything seems to be OK I'll take that as a "green light" to push out a release and y'all see a new _official_ release pop up...

**Comment by bguerin** [comment]
Hello,
latest snapshot of the plugin installed on my production system
seems all good, will let you known if I found something until the end of the week

my env : Debian bullseye host, Debian bullseye containers, Jenkins 2.357 running on docker with Java 11

**Comment by stephan48** [comment]
So far it looks quite good!

Need to test it further throu - not just running the same job a few times :P

as i need to pass in a single command without args I need to created a derivate windows image for now:
FROM jenkins/agent:jdk11-windowsservercore-ltsc2019
RUN mkdir c:\apps
COPY running.cmd c:/apps/running.cmd

running.cmd:
ping -t 127.0.0.1


for inbound-agent it looks like this:
FROM jenkins/inbound-agent:jdk11-windowsservercore-ltsc2019
RUN mkdir c:\apps
COPY running.cmd c:/apps/running.cmd

ENTRYPOINT ["c:/apps/running.cmd"]

running.cmd:
ping 127.0.0.1
set
echo "%*"
powershell.exe -f C:\ProgramData\Jenkins\jenkins-agent.ps1 %*

for inbound-agent we technically don't need to overwrite the image as the "connect via jnlp" option sets a reasonable command based on the container entrypoint. I just wanted to see which env vars we get :)

**Comment by Peter-Darton-i2** [comment]
So, it's been a week - Anyone have anything further to add?

@bguerin Would it be correct to say that, from your point of view, it's all working OK?

@stephan48 Were you able to "test it further"? Did it work ok?

(I'd like at least two people telling me "everything looks ok" before I push out a release)

**Comment by bguerin** [comment]
yes, from my point of view, it's all working OK

**Comment by CJCombrink** [comment]
I have finally gotten around to installing the latest snapshot and everything seems to be working. 
I will be testing for the rest of the day and we have some builds scheduled for over the weekend.

I will revert back on Monday on how testing went, but for now everything looks fine.

My versions for reference:
Jenkins 2.365
java.runtime.version: 11.0.16+8-LTS
docker-plugin: 1.2.10-rc1029.6bb665143e30

Docker cloud (Linux): Version = 20.10.17, API Version = 1.41
   - method: "Attach Docker Container"
Docker cloud (Windows): Version = 20.10.17, API Version = 1.41
    - method: "Connect with JNLP"

**Comment by CJCombrink** [comment]
Our builds are running without problem, daily and on demand builds. 
👍

**Comment by pjdarton** [comment]
Sorry for the delay - I must've missed that last notification.

I've done the release.
https://repo.jenkins-ci.org/releases/io/jenkins/docker/docker-plugin/1.2.10/ was uploaded a couple of hours ago.
